### PR TITLE
Create release.yml

### PR DESCRIPTION
**Pull request by kcowger**
Creating release.yml to make generating automated releases more informative.

## Product Description
Following directions here: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuring-automatically-generated-release-notes

This will make release notes much easier in the future. 

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by dannyroberts — APPROVED** [review]

**Comment by millerdev** [comment]
Does this put commit messages in release notes, or just PR titles? Might be worth notifying the team about this and reminding to use descriptive messages/titles as necessary to get high-quality release notes.

**Comment by kcowger** [comment]
> Does this put commit messages in release notes, or just PR titles? Might be worth notifying the team about this and reminding to use descriptive messages/titles as necessary to get high-quality release notes.

Thanks @millerdev , it's just going to use PR titles. I agree about moving toward more descriptive messages -- for now this is a way for me to quickly get a list of PRs by title, and then I'll be manually creating release notes to go on our wiki. In the future, I hope we can get to near full automation, though that will require team-wide behavior change around label and title usage. This is just step 1 of a few.

File: .github/release.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+changelog:
+  exclude:
+    authors:
+      - dependabot
+      - token-generator-for-cchq
+  categories:
+    - title: Product Updates
+      labels:
+        - product/all-users-all-environments
```
Comment (orangejenny) [inline .github/release.yml:11]: Should this also include `product/prod-india-all-users`?

**Review by orangejenny — APPROVED** [review]
This is fun.

File: .github/release.yml (hunk 1 above)
Comment (kcowger) [reply .github/release.yml:11]: Added! thanks for pointing that out

> Should this also include `product/prod-india-all-users`?

**Review by dannyroberts — APPROVED** [review]

**Pull request merged**